src/stage1/recurse.jl

```diff
@@ -276,7 +276,13 @@ function optic_transform!(ci, mi, nargs, N)
     stmts = VERSION < v"1.11.0-DEV.258" ? ir.stmts.inst : ir.stmts.stmt
```

Comment:
FWIW in the typical pipeline `convert_to_ircode` is used to convert CodeInfo-style statements to IRCode, but that's skipped here.

Most of the function is a no-op if you have no (informative) results from inference but there is some normalization (e.g. `strip_trailing_junk!`) that might be needed here for certain IR.

None of that is new in this PR though, so we can worry about it later.